src/app/address/controllers/address.js

```diff
@@ -3,23 +3,45 @@ const BaseController = require("hmpo-form-wizard").Controller;
 class AddressController extends BaseController {
   locals(req, res, callback) {
     super.locals(req, res, (err, locals) => {
-      locals.addressPostcode = req.sessionModel.get("addressPostcode");
-      const address = req.sessionModel.get("chosenAddress");
+      const addresses = req.sessionModel.get("addresses");
+      let address;
+
+      if (req.originalUrl === "/previous/address/edit") {
```

Comment:
If this gets any more complicated, I think we'll want to refactor this into three separate and smaller `form-wizards`:
- `currentAddress`
- `previousAddress`
- `summary`

as that should make the edit-and-return part of the journey work a lot more smoothly. (That's how the final page of the HMPO journey works).

That would also handle cases of two addresses, then adjusting the current address so that it is now more than 3 months ago (thus negating the need for the previous address).